v1/transform.go

```diff
@@ -0,0 +1,228 @@
+package v1
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+	v1Models "github.com/edgexfoundry/go-mod-core-contracts/v2/v1/models"
+)
+
+// TransformToV1DeviceProfile transform v2 profile to v1
+func TransformToV1DeviceProfile(profile models.DeviceProfile) v1Models.DeviceProfile {
+	dto := v1Models.DeviceProfile{
+		DescribedObject: v1Models.DescribedObject{},
+		Name:            profile.Name,
+		Manufacturer:    profile.Manufacturer,
+		Model:           profile.Model,
+		Labels:          profile.Labels,
+		DeviceResources: nil,
+		DeviceCommands:  nil,
+	}
+	dto.DeviceResources = toV1DeviceResources(profile.DeviceResources)
+	dto.DeviceCommands = toV1DeviceCommands(profile.DeviceCommands)
+	return dto
```

Comment:
you need a function to generate Core Command

Reply:
Added.